src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs

```diff
@@ -1650,6 +1652,33 @@ public async Task FormatArbitraryNodeParenthesizedLambdaExpression()
             await AssertFormatOnArbitraryNodeAsync(node, expected);
         }
 
+        [WorkItem(30518, "https://github.com/dotnet/roslyn/issues/30518")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public async Task FormatGeneratedNodeInInitializer()
+        {
+            var code = @"new int[] {
+    0,
+    1
+}";
+
+            var expected = @"new int[] {
+    0,
+Identifier Identifier,
+    1
+}";
+
+            var tree = SyntaxFactory.ParseSyntaxTree(code, options: TestOptions.Script);
+            var root = tree.GetRoot();
+
+            var declaration = SyntaxFactory.DeclarationExpression(SyntaxFactory.IdentifierName(SyntaxFactory.Identifier("Identifier")), SyntaxFactory.SingleVariableDesignation(SyntaxFactory.Identifier("Identifier")));
+
+            var newType = declaration.Type.WithoutTrivia().WithTrailingTrivia(SyntaxFactory.ElasticMarker).WithAdditionalAnnotations(Formatter.Annotation);
+            var strippedDeclaration = SyntaxFactory.DeclarationExpression(newType, declaration.Designation);
```

Comment:
declearation is already created from factory, so code below doesn't look like needed.